src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -199,6 +223,14 @@ class CoreWorkerDirectTaskSubmitter {
 
   // Retries cancelation requests if they were not successful.
   absl::optional<boost::asio::steady_timer> cancel_retry_timer_;
+
+
+  int pipeline_fullness_;
```

Comment:
I think we could call this something more descriptive, since not everyone will know what "pipeline fullness" means. Maybe `max_tasks_in_flight_per_worker_`? Also, could you make it `const` and add a comment describing what it is?